### ISSUE TITLE
About the cam_intrinsic after data augment

### ISSUE THREAD

**Issue by zengwz** [swc-17/SparseDrive#49]
sparsedrive/projects/mmdet3d_plugin/datasets/pipelines/augment.py
![image](https://github.com/user-attachments/assets/228bc47c-1fbe-4488-be54-713c3c053bf4)

Why the cam_intrinsic is fixed here by line 26 instead of line 27-29?

**Comment by swc-17** [comment]
Hi, what do you mean by "fixed"? The cam_intrinsic is changed by multiplying resize scale.

**Comment by zengwz** [comment]
Data augmentation includes resize, rotate, and crop. These should change the origin cam_intrinsic. line 26 only apply the resize to cam_intrinsic. It seems that line 27-29 is a more reasonable transformation? 

> Hi, what do you mean by "fixed"? The cam_intrinsic is changed by multiplying resize scale.

**Comment by swc-17** [comment]
Good point, we checked again, line 27-29 are correct transformation.

**Comment by zengwz** [comment]
Fine. Thanks for your reply.

**Comment by zengwz** [comment]
Based on my understanding, this will affect the values of gt_depth. I'm curious about how much the depth branch affects the final accuracy, which is not mentioned in the paper. Also, would adjusting gt_depth improve the final accuracy?

**Comment by swc-17** [comment]
I think this will not have significant impact, actually depth estimation is not necessary. See [issue](https://github.com/HorizonRobotics/Sparse4D/issues/6).

**Comment by zuochongyan** [comment]
for correctness,the true intrinsic should be 27~29,but in depth branch,the depth is rescaled by focal length(without abs), for flip augmentation,the focal length would be minus value, which causes wrong depth gt, i think a "correct" way is to use abs focal length to rescale in gt branch supervision, but currently it is same as line 26 because other positions in cam_intrinsic is not used such as u0,v0, but if you want to add intrinsic augmentation(add noise to intrinsic),line 27~29 should be used and use abs focal loss in gt depth rescale.

**Comment by swc-17** [comment]
> for correctness,the true intrinsic should be line 27-29,but in depth branch,the depth is rescaled by focal length(without abs), for flip augmentation,the focal length would be minus value, which causes wrong depth gt, i think a "correct" way is to use abs focal length to rescale in gt branch supervision, but currently it is same as line 26 because other positions in cam_intrinsic is not used such as u0,v0, but if you want to add intrinsic augmentation(add noise to intrinsic),line 27~29 should be used and use abs focal loss in gt depth rescale.

We can also compute focal as 
```    
data["focal"] = torch.tensor(
        np.sqrt(
            np.abs(np.linalg.det(data["cam_intrinsic"][:, :, :2, :2]))
        )
    )
```
